src/opm/io/eclipse/EclFile.cpp

```diff
@@ -616,96 +590,163 @@ void EclFile::loadArray(std::fstream& fileH, int arrIndex)
 
 void EclFile::loadData()
 {
-    std::fstream fileH;
 
     if (formatted) {
-        fileH.open(inputFilename, std::ios::in);
+
+        std::ifstream inFile(inputFilename);
```

Comment:
Reading the *whole file* into memory - unconditionally - is not a good solution; unified restart files which are larger than the available memory is not very special.

Reply:
Thanks for this comment. The largest unified formatted restart file I have found is 21 Gb. Normal field models are around 2.0 Gb. I see that 21 Gb is a lot of memory.

However I have tested an alternative where I divide reading from disk  into smaller chunks
```
void EclFile::loadData()so I will definitely 
{

    if (formatted) {

        std::vector<int> arrIndices(array_name.size());
        std::iota(arrIndices.begin(), arrIndices.end(), 0);

        this->loadData(arrIndices);
```

loadData(std::vector<int>& ) will read one array and process this at a time. 

This will actually reduce run time significantly (- 15%). I will definitely keep this change